infrastructure/gsuite-exporter/gsuite_exporter/exporters/stackdriver_exporter.py

```diff
@@ -0,0 +1,143 @@
+    """Convert Admin SDK logs to logging entries and send them to Stackdriver
+    Logging API.
+
+    Args:
+        api (`googleapiclient.discovery.Resource`): The Admin SDK API to fetch
+            records from.
+        version (str): The Admin SDK API version.
+        credentials_path (str, optional): The path to the GSuite Admin credentials.
+    """
+    SCOPES = [
+        'https://www.googleapis.com/auth/logging.read',
+        'https://www.googleapis.com/auth/logging.write'
+    ]
+    LOGGING_API_VERSION = 'v2'
+    def __init__(self,
+                 project_id,
+                 credentials_path=None):
+        logger.debug("Initializing Stackdriver Logging API ...")
+        self.api = auth.build_service(
+            api='logging',
+            version=StackdriverExporter.LOGGING_API_VERSION,
+            credentials_path=credentials_path,
+            scopes=StackdriverExporter.SCOPES)
+        self.project_id = "projects/{}".format(project_id)
+
+    def send(self, records, log_name, dry=False):
+        """Writes a list of Admin SDK records to Stackdriver Logging API.
+
+        Args:
+            records (list): A list of log records.
+            log_name (str): The log name to write (e.g: 'logins').
+            dry (bool): Toggle dry-run mode (default: False).
+
+        Returns:
+            `googleapiclient.http.HttpRequest`: The API response object.
+        """
+        res = None
+        destination = self.get_destination(log_name)
+        if records:
+            entries = self.convert(records)
+            body = {
+                'entries': entries,
+                'logName': '{}'.format(destination),
+                'dryRun': dry
+            }
+            logger.debug("Writing {} entries to Stackdriver Logging API @ '{}'".format(
+                len(entries),
+                destination))
+            res = self.api.entries().write(body=body).execute()
+        return res
+
+    def convert(self, records):
+        """Convert a bunch of Admin API records to Stackdriver Logging API
+        entries.
+
+        Args:
+            records (list): A list of Admin API records.
+
+        Returns:
+            list: A list of Stackdriver Logging API entries.
+        """
+        return map(lambda i: self.__convert(i), records)
+
+    def get_destination(self, log_name):
+        return "{}/logs/{}".format(self.project_id, log_name)
+
+    def get_last_timestamp(self, log_name):
+        """Last log timestamp from Stackdriver Logging API given our project id
+        and log name.
+        """
+        destination = self.get_destination(log_name)
+        query = {
+          'orderBy': 'timestamp desc',
+          'pageSize': 1,
+          'resourceNames': [self.project_id],
+          'filter': 'logName={}'.format(destination)
+        }
+        log = self.api.entries().list(body=query).execute()
+        timestamp = log['entries'][0]['timestamp'] if 'entries' in log else None
```

Comment:
Might this still result in an `IndexError` or `KeyError`?  Might be safer to wrap this in a `try...except`.

Reply:
I've changed that line to:
```
        try:
            timestamp = log['entries'][0]['timestamp']
        except (KeyError, IndexError):
            timestamp = None
```